superset/assets/javascripts/explorev2/components/ChartContainer.jsx

```diff
@@ -149,8 +158,13 @@ class ChartContainer extends React.Component {
     this.props.actions.removeChartAlert();
   }
 
-  renderVis() {
-    visMap[this.props.viz_type](this.state.mockSlice).render();
+  renderVis(heightChange) {
```

Comment:
it's confusing that the method is called `renderVis` but may not render the viz. Now that you're changing the method definition, the name isn't valid anymore. 

Personally what I would do is have a single lifecycle method (`componentWillReceiveProps `) handle all of this, no redirection. Just be careful with `this.state` not being instantaneous, use callbacks.